src/main/java/org/jilt/internal/AbstractBuilderGenerator.java

```diff
@@ -333,13 +335,34 @@ protected final TypeVariableName mangleTypeParameter(TypeVariableName typeVariab
     private ParameterSpec setterParameterSpec(VariableElement attribute, TypeName parameterType) {
         ParameterSpec.Builder ret = ParameterSpec.builder(parameterType,
                 this.attributeSimpleName(attribute));
-        AnnotationMirror nullableAnnotation = this.firstAnnotationCalledNullable(attribute);
-        if (nullableAnnotation != null) {
-            ret.addAnnotation(AnnotationSpec.get(nullableAnnotation));
-        }
+        this.addAnnotationsToParam(ret, attribute.getAnnotationMirrors());
+        this.addAnnotationsToParam(ret, attribute.asType().getAnnotationMirrors());
         return ret.build();
     }
 
+    private void addAnnotationsToParam(ParameterSpec.Builder ret, List<? extends AnnotationMirror> annotationMirrors) {
+        for (AnnotationMirror annotation : annotationMirrors) {
+            if (this.isAnnotationAllowedInParameter(annotation)) {
+                ret.addAnnotation(AnnotationSpec.get(annotation));
+            }
+        }
+    }
+
+    public boolean isAnnotationAllowedInParameter(AnnotationMirror annotation) {
```

Comment:
There are a few small issues with this method:

1. It shouldn't be `public`.
2. I think the correct word here is "on", not "in".
3. Let's use "param" instead of "parameter", to be consistent with the name above.

Putting that all together, this should be:

```suggestion
    private boolean isAnnotationAllowedOnParam(AnnotationMirror annotation) {
```

Reply:
copy pasting is the devil... haha. I like your thoroughness!